src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleasesController.cs

```diff
@@ -174,12 +174,39 @@ public async Task<ActionResult<FileInfo>> UpdateChartFileAsync(Guid releaseId, G
         [RequestSizeLimit(int.MaxValue)]
         [RequestFormLimits(ValueLengthLimit = int.MaxValue, MultipartBodyLengthLimit = int.MaxValue)]
         public async Task<ActionResult<DataFileInfo>> AddDataFilesAsync(Guid releaseId,
-            [FromQuery(Name = "name"), Required] string name, IFormFile file, IFormFile metaFile)
+            [FromQuery(Name = "name"), Required] string subjectName, IFormFile file, IFormFile metaFile)
         {
             var user = await _userManager.GetUserAsync(User);
 
             return await _releaseFilesService
-                .UploadDataFilesAsync(releaseId, file, metaFile, name, user.Email)
+                .UploadDataFilesAsync(
+                    releaseId: releaseId,
+                    dataFile: file,
+                    metaFile: metaFile,
+                    userName: user.Email,
+                    subjectName: subjectName)
+                .HandleFailuresOrOk();
+        }
+
+        [HttpPost("release/{releaseId}/data/replacing/{replacingId}")]
+        [Produces("application/json")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(404)]
+        [RequestSizeLimit(int.MaxValue)]
+        [RequestFormLimits(ValueLengthLimit = int.MaxValue, MultipartBodyLengthLimit = int.MaxValue)]
+        public async Task<ActionResult<DataFileInfo>> AddReplacementDataFilesAsync(Guid releaseId,
+            Guid replacingId,
```

Comment:
This would probably be better as `replacingFileId`.

Reply:
Thanks, I've done this now